airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +280,71 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    @staticmethod
+    def get_tickets(
+        result_return_limit: int, getter: Callable, params: Mapping[str, Any] = None, ticket_paginate_limit: int = 300
+    ) -> Iterator:
+        """
+        Read using getter
+
+        This block extends TicketsAPI Stream to overcome '300 page' server error.
+        Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        params = params or {}
+
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {
+            **params, 
+            "order_type": "asc", # ASC order, to get the old records first
+            "order_by": "updated_at", 
+            "per_page": result_return_limit
+        }
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < result_return_limit:
+                return iter(())
+
+            # checkpoint & switch the pagination
+            if page == ticket_paginate_limit:
+                # get last_record from latest batch, pos. -1, because of ACS order of records
+                last_record_updated_at = batch[-1]["updated_at"]
+                page = 0  # reset page counter
+                # Adding +1 second to last_record value, to avoid record duplication,
+                # this potentialy could lead to skipping some records during the fetch,
+                # in cases where multiple records are created at the same datetime, at the level of seconds
+                # this behaviour is tested in test_300_page.py unit_test (test1)
+                # to avoid such ocations, please comment or remove the `.add(seconds=1)' part.
+                last_record_updated_at = pendulum.parse(last_record_updated_at).add(seconds=1)
+                # updating request parameters with last_record state
+                params["updated_since"] = last_record_updated_at
+            # Increment page
+            page += 1
+
+    # Override the super().read() method with modified read for tickets
+    def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
```

Comment:
i believe this method can be replaced with: 

```
new_getter = partial(self.get_tickets, self, self._result_return_limit, getter, params)
yield from super().read(new_getter, params) 
```
some details might be wrong but the idea is to redefine the getter using the `get_tickets` method

Reply:
@sherifnada My, apology here, strugling with this new_getter and case to nowhere, it causes the infinity loop, when we redefine the new_getter, because the code is dependent of the 
`list()` function method, which produces `getter` for the `read` method, next, this `list` function is used in the `get_tickets` to produce the tickets stream output, for `read`, thus, i override the whole `read` in order to replace the original function with `get_tickets`. Otherwise it reads itself over and over again, in infinity loop. 

If you have any idea how to refactor the code in this case, probably the get_tickets. method is not perfect, I would leave it as it is, becase it's working as it should be. 

I was able to make this change:
`new_getter = partial(self.get_tickets, getter=partial(self._api_get, url="tickets"), result_return_limit=self.result_return_limit)
yield from self.read(getter=new_getter, params=params)`

Reply:
@bazarnov let's skipt he refactor